### PR TITLE
do not generate javadoc for tests

### PR DESCRIPTION
**Pull request by rkuhn**

### CODE REVIEW

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -313,7 +313,8 @@ object AkkaBuild extends Build {
     // compile options
     scalacOptions in Compile ++= Seq("-encoding", "UTF-8", "-target:jvm-1.8", "-feature", "-unchecked", "-Xlog-reflective-calls", "-Xlint"),
     scalacOptions in Compile ++= (if (allWarnings) Seq("-deprecation") else Nil),
-    scalacOptions in Test := (scalacOptions in Test).value.filterNot(_ == "-Xlog-reflective-calls"),
+    scalacOptions in Test := (scalacOptions in Test).value.filterNot(opt =>
+      opt == "-Xlog-reflective-calls" || opt.contains("genjavadoc")),
```
Comment (patriknw) [inline project/AkkaBuild.scala]: alright, magic

**Comment by patriknw** [comment]
LGTM, thanks

File: project/AkkaBuild.scala (hunk 1 above)
Comment (rkuhn) [inline project/AkkaBuild.scala]: No, not at all: we previously added the configuration that emits the synthetic Java sources to the Test configuration, now we don’t.

Comment (patriknw) [inline project/AkkaBuild.scala]: where do we add this opt to scalacOptions in the first place?
I can't find the string "genjavadoc" anywhere else than `scalacOptions in Compile += "-P:genjavadoc:fabricateParams=true"`, but that can't be the thing.
Is it added by unidoc?

Comment (rkuhn) [inline project/AkkaBuild.scala]: Yes, it is in genjavadocExtraSettings, pulled in from `project/Unidoc.scala`.

**Pull request merged**